src/julia_to_gap.jl

```diff
@@ -133,7 +143,7 @@ function julia_to_gap(
         end
         if recursive
             x = get!(recursion_dict, x) do
```

Comment:
The compiler can not infer the type for `recursion_dict` here, but we can
```suggestion
            x = get!(recursion_dict::RecDict, x) do
```

Reply:
Oh, and an existing issue: we should not use `x` both for the original value for the converted value. So it would be good to change `x = ` here to e.g. `res = ...` and then below change `ret_val[i] = x` o `ret_val[i] = res` (or whatever)